src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1413,7 +1416,112 @@ private IBlockOperation CreateBoundBlockOperation(BoundBlock boundBlock)
+                    isLifted: false,
+                    isChecked: false,
+                    isCompareText: false,
+                    operatorMethod: null,
+                    unaryOperatorMethod: null,
+                    _semanticModel,
+                    syntax,
+                    compilation.GetSpecialType(SpecialType.System_Boolean),
+                    constantValue,
+                    isImplicit: true);
+            }
+
+            var argumentNullExceptionMethodSymbol = (IMethodSymbol)compilation.GetWellKnownTypeMember(WellKnownMember.System_ArgumentNullException__ctorString);
+            var argumentNullExceptionType = compilation.GetWellKnownType(WellKnownType.System_ArgumentNullException);
+            var argumentNullExceptionObject = new ObjectCreationOperation(
+                        constructor: argumentNullExceptionMethodSymbol,
+                        initializer: new ObjectOrCollectionInitializerOperation(
+                                initializers: ImmutableArray<IOperation>.Empty,
+                                _semanticModel,
+                                syntax,
+                                compilation.GetWellKnownType(WellKnownType.System_ArgumentNullException),
+                                constantValue,
+                                isImplicit: true),
+                        ImmutableArray.Create<IArgumentOperation>(
+                            new ArgumentOperation(
+                                new LiteralOperation(_semanticModel, syntax, compilation.GetSpecialType(SpecialType.System_String), parameter.Name, isImplicit: true),
+                                ArgumentKind.Explicit,
+                                parameter: argumentNullExceptionMethodSymbol.Parameters[0],
+                                inConversionOpt: null,
+                                outConversionOpt: null,
+                                _semanticModel,
+                                syntax,
+                                isImplicit: true)),
+                        _semanticModel,
+                        syntax,
+                        type: argumentNullExceptionType,
+                        constantValue,
+                        isImplicit: true);
+
+            IOperation whenTrue = new ExpressionStatementOperation(
+                new ThrowOperation(
+                    argumentNullExceptionObject,
+                    _semanticModel,
+                    syntax,
+                    type: argumentNullExceptionType,
+                    constantValue,
+                    isImplicit: true),
+                _semanticModel,
+                syntax,
+                compilation.GetSpecialType(SpecialType.System_Void),
+                constantValue,
+                isImplicit: true);
+            return new ConditionalOperation(conditionOp, whenTrue, null, isRef: false, _semanticModel, syntax, compilation.GetSpecialType(SpecialType.System_Boolean), constantValue, isImplicit: true);
```

Comment:
>null [](start = 67, length = 4)

Consider using a named parameter here.

Reply:
For void or the bool type? If bool, done and pushed.

Reply:
For the null literal

---
In reply to: [305931912](https://github.com/dotnet/roslyn/pull/37381#discussion_r305931912) [](ancestors = 305931912)